app/src/main/java/com/example/fitjournal/home/presentation/components/card/WeightLiftingCard.kt

```diff
@@ -66,39 +73,48 @@ private fun TopSetTitle() {
     Text(
         text = stringResource(id = R.string.title_top_set),
         style = MaterialTheme.typography.titleMedium,
+        color = MaterialTheme.colorScheme.onPrimary,
         modifier = Modifier.padding(horizontal = Spacing.spacing16)
     )
 }
 
 @Composable
 private fun TopSetSummary(
-    reps: Int?,
-    sets: Int?,
-    weight: Double?,
+    weightLiftingUi: WeightLiftingUi?,
     modifier: Modifier = Modifier
 ) {
     Column(
         modifier = modifier,
         verticalArrangement = Arrangement.SpaceBetween,
         horizontalAlignment = Alignment.Start
     ) {
-        sets?.let {
+        if (weightLiftingUi?.sets != null &&
+            weightLiftingUi.reps != null &&
+            weightLiftingUi.weight != null
+        ) {
```

Comment:
I left a comment about moving null checks to an extension function inside data class, we can do that here